bin/send_cmd.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+
+################################################################################
+# This script is used to send a command to the node with AWS SSM
+# commands are cluster wide, it will self pick the target nodes.
+#
+# supported commands:
+#  - start_traffic
+#     start background traffic on the node (pub and sub)
+#  - collect_logs
+#     collect logs from the emqx nodes and copy it to S3 bucket
+################################################################################
+
+
+# cluster name 'CDK_EMQX_CLUSTERNAME' when you deploy cdk
+cluster=$1
+
+# CdkChaosTests.ControlCmd,
+# scripts are under cdk_emqx_cluster/ssm_docs/
+command=$2
+
+die() {
+    echo $1
+    exit 1;
+}
+
+if [ -z "$cluster" ]; then
+    die "cluster not set";
+fi
+
+doc_name=$(aws ssm list-documents --filters "Key=tag:cluster,Values=$cluster" --filters "Key=tag:cmd,Values=$command" | jq -r '.DocumentIdentifiers[0].Name')
+if [[ "$doc_name" == "null" || ! $? ]]; then
+    die "command '$command' not found, did you deploy the CdkChaosTests Stack?"
+fi
+
+case $command in
+    "start_traffic")
+        # supply command parameters and targets
+        LB="lb.int.$cluster"
+        aws ssm send-command --document-name "$doc_name" \
+            --parameters "{\"Host\": [\"$LB\"], \"Command\":[\"sub\"],\"Prefix\":[\"cdk\"],\"Topic\":[\"topic_a\"],\"Clients\":[\"200000\"],\"Interval\":[\"50\"]}" \
+            --targets "[{\"Key\":\"tag:cluster\",\"Values\":[\"$cluster\"]},{\"Key\":\"tag:service\",\"Values\":[\"loadgen\"]}]" \
+            --timeout-seconds 60 --max-concurrency "30" --max-errors "0"
+
+        aws ssm send-command --document-name "$doc_name" \
+            --parameters "{\"Host\": [\"$LB\"], \"Command\":[\"pub\"],\"Prefix\":[\"cdk\"],\"Topic\":[\"topic_a\"],\"Clients\":[\"200000\"],\"Interval\":[\"50\"]}" \
+            --targets "[{\"Key\":\"tag:cluster\",\"Values\":[\"$cluster\"]},{\"Key\":\"tag:service\",\"Values\":[\"loadgen\"]}]" \
+            --timeout-seconds 60 --max-concurrency "30" --max-errors "0"
+        ;;
+    "collect_logs")
+        aws ssm send-command --document-name "$doc_name" \
+            --parameters "{\"Bucket\":[\"emqx-cdk-cluster\"],\"Path\":[\"$cluster\"]}" \
```

Comment:
Curiosity: should there be a `Command` key in those parameters, like `pub` and `sub` above?

Reply:
If the object path is only the cluster name, subsequent runs will overwrite previous logs, right?  Would it make sense to namespace the experiment run with some unique identifier, like the start datetime or similar?

Reply:
You are right, node_dump  generates a unique log filename. Also if we integrate with CI, we could use the job name as the cluster name.